src/proxy/outbound.rs

```diff
@@ -363,91 +320,54 @@ impl OutboundConnection {
+        let state = &self.pi.state;
+        // First find the source workload of this traffic. If we don't know where the request is from
+        // we will reject it.
+        let source_workload = self.fetch_source_workload(downstream).await?;
 
         // If this is to-service traffic check for a service waypoint
-        // Capture result of whether or not this is svc addressed
-        let svc_addressed = if let Some(Address::Service(s)) = self
-            .pi
-            .state
-            .fetch_destination(&Destination::Address(NetworkAddress {
-                network: strng::new(&self.pi.cfg.network),
+        // Capture result of whether this is svc addressed
+        let svc_addressed = if let Some(Address::Service(target_service)) = state
+            .fetch_address(&NetworkAddress {
+                network: self.pi.cfg.network.clone(),
                 address: target.ip(),
-            }))
+            })
             .await
         {
             // if we have a waypoint for this svc, use it; otherwise route traffic normally
-            if let Some(wp) = s.waypoint.clone() {
-                let waypoint_vip = match wp.destination {
-                    Destination::Address(a) => a.address,
-                    Destination::Hostname(_) => {
-                        return Err(proxy::Error::UnknownWaypoint(
-                            "hostname lookup not supported yet".to_string(),
-                        ));
-                    }
-                };
-                let waypoint_vip = SocketAddr::new(waypoint_vip, wp.hbone_mtls_port);
-                let waypoint_us = self
-                    .pi
-                    .state
-                    .fetch_upstream(self.pi.cfg.network.clone(), &source_workload, waypoint_vip)
-                    .await
-                    .ok_or(proxy::Error::UnknownWaypoint(
-                        "unable to determine waypoint upstream".to_string(),
-                    ))?;
-
-                let waypoint_workload = waypoint_us.workload;
-                let waypoint_ip = self
-                    .pi
-                    .state
-                    .pick_workload_destination(
-                        &waypoint_workload,
-                        &source_workload,
-                        self.pi.metrics.clone(),
-                    )
-                    .await?; // if we can't load balance just return the error
-
-                let waypoint_socket_address = SocketAddr::new(waypoint_ip, waypoint_us.port);
-                let id = waypoint_workload.identity();
-                return Ok(Box::new(Request {
+            if let Some(waypoint) = state
+                .fetch_service_waypoint(&target_service, &source_workload)
+                .await?
+            {
+                let upstream_sans = waypoint.workload_and_services_san();
+                let actual_destination = waypoint.workload_socket_addr();
+                trace!("built request to service waypoint proxy");
```

Comment:
```suggestion
                debug!("built request to service waypoint proxy");
```

Mega nit: slight pref for logs indicating request processing decisions to be `debug!` - trace is a lot  of noise.